scripts/tests/test_scripts.py

```diff
@@ -233,127 +234,43 @@ def test_pretrain_create():
-                                         '--warmup_ratio', '0.5',
-                                         '--pretrained'] + arguments)
-        # test training with raw data
-        process = subprocess.check_call([sys.executable, './scripts/bert/run_pretraining_hvd.py',
-                                         '--dtype', 'float32',
-                                         '--raw',
-                                         '--max_seq_length', '128',
-                                         '--max_predictions_per_seq', '20',
-                                         '--masked_lm_prob', '0.15',
-                                         '--short_seq_prob', '0.1',
-                                         '--data', './scripts/bert/sample_text.txt',
-                                         '--data_eval', './scripts/bert/sample_text.txt',
-                                         '--batch_size', '32',
-                                         '--lr', '2e-5',
-                                         '--warmup_ratio', '0.5',
-                                         '--pretrained'] + arguments)
-
-        # test evaluation
-        process = subprocess.check_call([sys.executable, './scripts/bert/run_pretraining_hvd.py',
-                                         '--dtype', 'float32',
-                                         '--data_eval', './test/bert/data/*.npz',
-                                         '--eval_use_npz', '--pretrained'] + arguments)
-
-        # test mixed precision training and use-avg-len
-        from mxnet.ndarray.contrib import mp_adamw_update
-        process = subprocess.check_call([sys.executable, './scripts/bert/run_pretraining_hvd.py',
-                                         '--data', './test/bert/data/*.npz',
-                                         '--data_eval', './test/bert/data/*.npz',
-                                         '--batch_size', '4096',
-                                         '--use_avg_len',
-                                         '--lr', '2e-5',
-                                         '--warmup_ratio', '0.5',
-                                         '--pretrained'] + arguments)
-        time.sleep(5)
-    except ImportError:
-        print("The test expects master branch of MXNet and Horovod. Skipped now.")
+    arguments = ['--log_interval', '2',
+                 '--lr', '2e-5', '--warmup_ratio', '0.5',
+                 '--total_batch_size', '32', '--total_batch_size_eval', '8',
+                 '--ckpt_dir', './test/bert/ckpt',
+                 '--num_steps', '20', '--num_buckets', '1',
+                 '--pretrained',
+                 '--comm_backend', backend]
+
+    if backend == 'horovod':
+        try:
+            # Test only if horovod is present
+            import horovod.mxnet as hvd
+        except ImportError:
+            print("The test expects master branch of MXNet and Horovod. Skipped now.")
```

Comment:
Can we execute this test on our CI? The time-stamps / execution time for both `master-gpu-integration` and `gpu-integration` suggest that it is skipped for both.

Reply:
The test with 'device' option should be execute (it's now appearing in the list of slowest tests). I tried to add horovod dependency on CI (https://github.com/dmlc/gluon-nlp/pull/775/files) but I failed with multipled attempts. Unfortunately the log is gone and I cannot find the exact error msg..

Reply:
Can you remove the try / except block given that horovod is now working on CI? (based on your latest commit and cuda upgrade on CI)